src/main/java/org/snomed/aag/data/services/CriteriaItemService.java

```diff
@@ -165,4 +170,54 @@ public void verifyManual(CriteriaItem criteriaItem, boolean expectedManual) {
 			throw new AccessDeniedException("Criteria Item cannot be changed manually.");
 		}
 	}
+
+	/**
+	 * Remove entry from collection. If entry does not have a corresponding flag in Branch metadata
+	 * or the corresponding flag in Branch metadata is false, then remove entry from collection.
+	 *
+	 * @param criteriaItems Collection to process
+	 * @param branch        Branch to cross reference
+	 */
+	public void removeNonEnabled(Set<CriteriaItem> criteriaItems, Branch branch) {
+		verifyParams(criteriaItems, branch);
+		Set<String> authorFlags = MetadataUtil.getEnabledAuthorFlags(branch).keySet();
+		if (!authorFlags.isEmpty()) {
+			criteriaItems.removeIf(item -> item.getEnabledByFlag().stream().noneMatch(authorFlags::contains));
+		}
+	}
```

Comment:
I originally had the name of the method as `removeDisabled` but I felt that was confusing as both disabled and not present flags will be removed. I also debated using terms "hide" and "delete". "Remove" felt more appropriate. 

This method is used when viewing the PAC for a branch.